File: app/src/main/java/com/dumuzeyn/mp3player/GenresMenuRenderer.java
package com.dumuzeyn.mp3player;

import java.util.ArrayList;
import java.util.Map;

final class GenresMenuRenderer extends TrackGroupMenuRenderer {
    GenresMenuRenderer(MainActivityCore host) {
        super(host);
    }

    @Override
    Map<String, ArrayList<Track>> groupedTracks() {
        return host.libraryState.homeContent.genreTracks;
    }

    @Override
    String unknownGroupName() {
        return host.tr("Unknown genre", "Неизвестный жанр");
    }

    @Override
    int cardOpacity() {
        return host.appearanceState.genreCardOpacity;
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/GenresMenuRenderer.kt
package com.dumuzeyn.mp3player

internal class GenresMenuRenderer(host: MainActivityCore) : TrackGroupMenuRenderer(host) {
    override fun groupedTracks(): Map<String, ArrayList<Track>> =
        host.libraryState.homeContent.genreTracks

    override fun unknownGroupName(): String = host.tr("Unknown genre", "Неизвестный жанр")

    override fun cardOpacity(): Int = host.appearanceState.genreCardOpacity
}
